File: Core/GrowthWare.DataAccess/SQLServer/DDBInformation.cs
using GrowthWare.DataAccess.Interfaces;
using GrowthWare.DataAccess.SQLServer.Base;
using GrowthWare.Framework.Models;
using System.Data;
using System.Data.SqlClient;

namespace GrowthWare.DataAccess.SQLServer
{
    public class DDBInformation : DDBInteraction, IDBInformation
    {
        private MDBInformation m_Profile = null;
        public MDBInformation Profile { get{return this.m_Profile;} set{this.m_Profile = value;} }

        public DataRow GetProfileRow
        {
            get
            {
                string mStoredProcedure = "ZGWSystem.Get_Database_Information";
                return base.GetDataRow(mStoredProcedure);
            }
        }

        public bool UpdateProfile()
        {
            string mStoredProcedure = "ZGWSystem.Set_DataBase_Information";
            SqlParameter[] mParameters = { 
                new SqlParameter("@P_Database_InformationSeqId", m_Profile.InformationSeqId),
                new SqlParameter("@P_Version", m_Profile.Version),
                new SqlParameter("@P_Enable_Inheritance", m_Profile.EnableInheritance),
                new SqlParameter("@P_Added_Updated_By", GetAddedUpdatedBy(m_Profile)),

                GetSqlParameter("@P_Primary_Key", -1, ParameterDirection.InputOutput)
            };
            try
            {
                 base.ExecuteNonQuery(mStoredProcedure, mParameters);
                 return true;
            }
            catch (System.Exception)
            {
                throw;
            }
        }
    }
}

File: Net_Framework/VB/GrowthWare.Framework.BusinessData/DataAccessLayer/SQLServer/V2008/DDBInformation.vb
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.SQLServer.Base
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces
Imports GrowthWare.Framework.Model.Profiles
Imports System.Data.SqlClient

Namespace DataAccessLayer.SQLServer.V2008
    <System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId:="DDB")> Public Class DDBInformation
        Inherits DDBInteraction
        Implements IDBInformation

        Private m_Profile As MDBInformation = Nothing

        Public ReadOnly Property GetProfileRow() As DataRow Implements IDBInformation.GetProfileRow
            Get
                Dim mStoredProcedure As String = "ZGWSystem.Get_Database_Information"
                Return MyBase.GetDataRow(mStoredProcedure)
            End Get
        End Property

        Public Property Profile As MDBInformation Implements IDBInformation.Profile
            Get
                Return m_Profile
            End Get
            Set(ByVal value As MDBInformation)
                m_Profile = value
            End Set
        End Property

        Public Function UpdateProfile() As Boolean Implements IDBInformation.UpdateProfile
            Dim mStoredProcedure As String = "ZGWSystem.Set_DataBase_Information"
            Dim mParameters() As SqlParameter = {
                New SqlParameter("@P_Database_Information_SeqID", m_Profile.InformationSeqId),
                New SqlParameter("@P_Version", m_Profile.Version),
                New SqlParameter("@P_Enable_Inheritance", m_Profile.EnableInheritance),
                New SqlParameter("@P_Added_Updated_By", GetAddedUpdatedBy(m_Profile)),
                GetSqlParameter("@P_Primary_Key", -1, ParameterDirection.InputOutput)
             }
            Try
                MyBase.ExecuteNonQuery(mStoredProcedure, mParameters)
                Return True
            Catch ex As Exception
                Throw
            End Try

        End Function
    End Class

End Namespace
